platform/mbed_error.h

```diff
@@ -66,12 +845,220 @@
  *
  *
  */
+    
+MBED_DEPRECATED_SINCE("mbed-os-5.9", "This function has been deprecated, please use one of SET_WARNING/SET_ERROR macros or one of set_warning/set_error functions" )
 
-#ifdef __cplusplus
-extern "C" {
-#endif
 void error(const char* format, ...);
```

Comment:
We still should not deprecate `error`

Reply:
Can I know why? We are planning to raise requirements for other modules to adopt new error handling. And this needs to be deprecated now before they start doing it.